lambda/src/main/scala/pricemigrationengine/services/CohortSpecTableLive.scala

```diff
@@ -0,0 +1,63 @@
+package pricemigrationengine.services
+
+import java.time.LocalDate
+
+import com.amazonaws.services.dynamodbv2.model.{AttributeValue, ScanRequest}
+import pricemigrationengine.model._
+import zio.{IO, ZIO, ZLayer}
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object CohortSpecTableLive {
+
+  private val tableNamePrefix = "price-migration-engine-cohort-spec"
+
+  private def toCohortSpec(values: mutable.Map[String, AttributeValue]): Either[CohortSpecFetchFailure, CohortSpec] = {
+
+    def stringFieldValue(fields: mutable.Map[String, AttributeValue],
+                         fieldName: String): Either[CohortSpecFetchFailure, String] =
+      fields.get(fieldName).map(_.getS).toRight(CohortSpecFetchFailure(s"Can't read field '$fieldName'"))
+
+    def optStringFieldValue(fields: mutable.Map[String, AttributeValue], fieldName: String): Option[String] =
+      fields.get(fieldName).map(_.getS)
+
+    for {
+      cohortName <- stringFieldValue(values, "cohortName")
+      earliestPriceMigrationStartDate <- stringFieldValue(values, "earliestPriceMigrationStartDate").map(
+        LocalDate.parse)
+      importStartDate <- stringFieldValue(values, "importStartDate").map(LocalDate.parse)
```

Comment:
Should we explicitly handle the potential parsing failure here.

Also there are a bunch of these conversion functions in the CohortTableLive, maybe it would be worth moving them to shared code?

Reply:
Yes for both.  I'll update the PR